cwms-data-api/src/main/java/cwms/cda/data/dao/LocationGroupDao.java

```diff
@@ -215,6 +215,85 @@ public List<LocationGroup> getLocationGroups(@Nullable String officeId,
+     * as well as a where clause to filter the shared_ref_location_id.
+     * @Param officeId The office id to use for the query.
+     * @Param locCategoryLike A regex to use to filter the location categories.  May be null.
+     * @Param sharedRefLocLike A where clause to filter the shared_loc_alias_id.  May be null.
+     * @return A list of all location groups for the given parameters.
+     */
+
+    public List<LocationGroup> getLocationGroups(String officeId, String locCategoryLike, String sharedRefLocLike) {
+        AV_LOC_GRP_ASSGN alga = AV_LOC_GRP_ASSGN.AV_LOC_GRP_ASSGN;
+        AV_LOC_CAT_GRP alcg = AV_LOC_CAT_GRP.AV_LOC_CAT_GRP;
+
+        final RecordMapper<Record, Pair<LocationGroup, AssignedLocation>> mapper = grpRecord -> {
+            LocationCategory category = buildLocationCategory(grpRecord);
+
+            LocationGroup group = buildLocationGroup(grpRecord, category);
+            AssignedLocation loc = buildAssignedLocation(grpRecord);
+
+            return new Pair<>(group, loc);
+        };
+
+        Condition condition = noCondition();
+        if (locCategoryLike != null && !locCategoryLike.isEmpty()) {
+            condition = caseInsensitiveLikeRegex(alcg.LOC_CATEGORY_ID, locCategoryLike);
+        }
+
+        if (sharedRefLocLike != null && !sharedRefLocLike.isEmpty()) {
+            condition = condition.and(caseInsensitiveLikeRegex(alcg.SHARED_REF_LOCATION_ID, sharedRefLocLike));
+        }
+
+        condition = condition.and(alcg.LOC_GROUP_ID.isNotNull());
+
+        SelectConnectByStep<? extends Record> connectBy;
+        SelectOnConditionStep<? extends Record> onStep = dsl.select(
+                        alcg.CAT_DB_OFFICE_ID,
+                        alcg.LOC_CATEGORY_ID,
+                        alcg.LOC_CATEGORY_DESC,
+                        alcg.GRP_DB_OFFICE_ID,
+                        alcg.LOC_GROUP_ID,
+                        alcg.LOC_GROUP_DESC,
+                        alcg.LOC_GROUP_ATTRIBUTE,
+                        alcg.SHARED_LOC_ALIAS_ID,
+                        alcg.SHARED_REF_LOCATION_ID,
+                        alga.DB_OFFICE_ID,
+                        alga.LOCATION_ID,
+                        alga.ALIAS_ID,
+                        alga.ATTRIBUTE,
+                        alga.REF_LOCATION_ID)
+                .from(alcg).leftJoin(alga)
+                .on(alcg.LOC_CATEGORY_ID.eq(alga.CATEGORY_ID)
+                        .and(alcg.LOC_GROUP_ID.eq(alga.GROUP_ID)));
+
+        if (officeId != null) {
+            connectBy = onStep.where(DSL.upper(alcg.GRP_DB_OFFICE_ID).eq(officeId.toUpperCase()).and(condition));
+        } else {
+            connectBy = onStep.where(condition);
+        }
+
+
+        Map<LocationGroup, List<AssignedLocation>> map = new LinkedHashMap<>();
+        connectBy.orderBy(alcg.LOC_CATEGORY_ID, alcg.LOC_GROUP_ID, alga.ATTRIBUTE)
+                .fetchSize(1000)  // This made the query go from 2 minutes to 10 seconds?
```

Comment:
That's likely to be rather query specific as it let's Oracle decide how much to gather and prepare at once as the downstream system calls `ResultSet::next`

Reply:
I believe it remains constant between queries. I can turn it into a constant since it's also used in the function that I based this one off of. It seems Ryan R originally added this optimization about a year ago, and I didn't want to break it if it was still necessary.

Reply:
https://github.com/search?q=repo%3AUSACE%2Fcwms-data-api%20fetchSize&type=code
Shows constants being 1000 vs 500, with default use for most cases I suspect. Default for Oracle appears to be 10:
https://docs.oracle.com/en/database/oracle/oracle-database/23/jjdbc/resultset.html#GUID-BD27B43E-525F-44C4-973F-18D2BB8BA007
10 is too small and that should be set in the connection pool to a better number. The 500 vs 1000 could be captured in a global function to allow variation or as global constants to allow easy change. Having them as int local values makes maintenance difficult.

Reply:
I'll look into making a global variable for this somewhere that is available to all the DAOs that use it. I'm open to suggestions for where to put it, but I'll try to find a helper file or base class that makes sense for this.

Reply:
I added a couple global variables in the JooqDao class, since that is shared between the DAOs that use fetchSize. Take a look at the PR here: https://github.com/USACE/cwms-data-api/pull/877